tests/array.rs

```diff
@@ -2002,6 +2002,35 @@ fn test_map_axis() {
     let c = a.map_axis(Axis(1), |view| view.sum());
     let answer2 = arr1(&[6, 15, 24, 33]);
     assert_eq!(c, answer2);
+
+    // Test zero-length axis case
+    let arr = Array3::<f32>::zeros((3, 0, 4));
+    let mut counter = 0;
+    let result = arr.map_axis(Axis(1), |x| {
+        assert_eq!(x.shape(), &[0]);
+        counter += 1;
+        counter
+    });
+    assert_eq!(result.shape(), &[3, 4]);
+    assert_eq!(result, arr2(&[
```

Comment:
Overall, this test is good, but we don't guarantee the iteration order of `map_axes`, so the elements of `result` are not guaranteed to be arranged in this way. Something like

```rust
itertools::assert_equal(result.iter().cloned().sorted(), 1..=3*4);
```

would work to verify that the elements are correct while ignoring their ordering.

(For access to `.sorted()`, make sure to `use itertools::Itertools` and add `features = ["use_std"]` to the `itertools` dependency in `Cargo.toml`.)

Reply:
Thanks, I've updated the test accordingly.